pandas/tests/io/parser/common/test_common_basic.py

```diff
@@ -862,3 +862,19 @@ def test_read_seek(all_parsers):
             actual = parser.read_csv(file)
         expected = parser.read_csv(StringIO(content))
     tm.assert_frame_equal(actual, expected)
+
+
+def test_pyarrow_with_multi_character_separator():
+    # GH#52554
+    import re
+
+    import pandas as pd
+
+    data = "a b c\n1 2 3\n4 5 6"
+    msg = re.escape(
+        "the 'pyarrow' engine does not support separators "
+        "with more than one character, except for '\\t'. Please "
+        "use the C parser instead."
+    )
+    with pytest.raises(ValueError, match=msg):
+        pd.read_csv(StringIO(data), engine="pyarrow", sep=r"\s+")
```

Comment:
It'd be good to make the separator a pytes parametrized value (adding a pytest.mark.parametrize decorator to the test iirc).

Then, provide other values that should fail too. I guess `  ` (two spaces) should also fail, as well as maybe something like `||`

Reply:
Done